test/e2e/thv-operator/virtualmcp/virtualmcp_yardstick_base_test.go

```diff
@@ -243,4 +244,118 @@ var _ = Describe("VirtualMCPServer Yardstick Base", Ordered, func() {
 			Expect(vmcpServer.Status.Phase).To(Equal(mcpv1alpha1.VirtualMCPServerPhaseReady))
 		})
 	})
+
+	Context("when testing group membership changes trigger reconciliation", func() {
+		backend3Name := "yardstick-c"
+
+		It("should have two discovered backends initially", func() {
+			status, err := GetVirtualMCPServerStatus(ctx, k8sClient, vmcpServerName, testNamespace)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(status.BackendCount).To(Equal(2), "Should have 2 initial backends")
+			Expect(status.DiscoveredBackends).To(HaveLen(2), "Should have 2 discovered backends")
+
+			backendNames := make([]string, len(status.DiscoveredBackends))
+			for i, backend := range status.DiscoveredBackends {
+				backendNames[i] = backend.Name
+			}
+			Expect(backendNames).To(ContainElements(backend1Name, backend2Name))
+
+			By(fmt.Sprintf("Initial backends: %v", backendNames))
+		})
+
+		It("should discover a new backend when added to the group", func() {
+			By("Creating a new yardstick backend MCPServer and adding to the group")
+			CreateMCPServerAndWait(ctx, k8sClient, backend3Name, testNamespace,
+				mcpGroupName, images.YardstickServerImage, timeout, pollingInterval)
+
+			By("Waiting for VirtualMCPServer to reconcile and discover the new backend")
+			Eventually(func() error {
+				status, err := GetVirtualMCPServerStatus(ctx, k8sClient, vmcpServerName, testNamespace)
+				if err != nil {
+					return err
+				}
+
+				if status.BackendCount != 3 {
+					return fmt.Errorf("expected 3 backends, got %d", status.BackendCount)
+				}
+
+				if len(status.DiscoveredBackends) != 3 {
+					return fmt.Errorf("expected 3 discovered backends, got %d", len(status.DiscoveredBackends))
+				}
+
+				backendNames := make([]string, len(status.DiscoveredBackends))
+				for i, backend := range status.DiscoveredBackends {
+					backendNames[i] = backend.Name
+				}
+
+				if !slices.Contains(backendNames, backend3Name) {
+					return fmt.Errorf("new backend %s not found in discovered backends: %v", backend3Name, backendNames)
+				}
+
+				return nil
+			}, timeout, pollingInterval).Should(Succeed(), "VirtualMCPServer should discover the new backend")
+
+		})
+
+		It("should remove a backend when deleted from the group", func() {
+			By("Deleting a backend MCPServer from the group")
+			backend2 := &mcpv1alpha1.MCPServer{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      backend2Name,
```

Comment:
This might be too much of a nitpick, but if are we deleting a server used by other tests? Wouldnt' it be better to have a server specifically for adding and deleting in this test?